metadata-ingestion-modules/airflow-plugin/src/datahub_airflow_plugin/client/airflow_generator.py

```diff
@@ -208,6 +209,7 @@ def generate_datajob(
         set_dependencies: bool = True,
         capture_owner: bool = True,
         capture_tags: bool = True,
+        **kwargs: Any,
```

Comment:
How come you opted for parsing kwargs rather than making `config` an explicit param? Parsing kwargs has some drawbacks: callers may forget to pass `config` when they should, and more generally the method is less well documented on what you should pass in

Reply:
Agree with Andrew here - I'd prefer to explicitly pass in the arg here

Reply:
@asikowitz @hsheth2 
It is good to have `**kwargs` if the function needs to be extensible so people do not need to refactor the code again and again. If the caller does not pass, it will use the default link, which is what you guys have now.

Reply:
In this particular case, I don't like using `**kwargs` - a big benefit of making it explicit is that we get type checking for each argument, and won't run into issues subtle issues with too many / missing arguments

I'd also prefer to keep the convention of passing individual keywords instead of passing the full config object for the `airflow_generator.py` class, since it is used in a number of different places and not all of them have the same config type

I would agree that it is getting a bit repetitive. Longer term, I'd like to change the AirflowGenerator class to accept settings on initialization e.g. `config.make_airflow_generator().generate_datajob()`, where make_airflow_generator would be `return AirflowGenerator(capture_tags=..., ...)`. But we don't need to do that right now

Reply:
hmm, if you take a look at other projects, adding `**kwargs` is quite common and flexible. If the fields are important, its better to list them out explicitly. This type of things is quite subjective  though.

Can you say any other configs right now generate_datajob can use? 

I can update the code again in the interest of time.